### PR TITLE
Improves react-navigation typings

### PR DESCRIPTION
**Pull request by chagasaway**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by lijunle** [comment]
@chagasaway I checked that react-navigation@3 has been [released](https://reactnavigation.org/blog/2018/11/17/react-navigation-3.0.html) today. Is the change still compatible with the new version?

**Comment by chagasaway** [comment]
@lijunle those changes are compatible with the 2.x version which is the currently supported here. I can help with the typings for the new version later but those changes from this PR aims only to improve the 2.x version instead of making it compatible with the 3.x

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -129,6 +133,8 @@ export interface NavigationLeafRoute<Params = NavigationParams> {
    * e.g. `{ car_id: 123 }` in a route that displays a car.
    */
   params?: Params;
+  routes: NavigationRoute[];
```
Comment (lijunle) [inline types/react-navigation/index.d.ts]: Add JSDoc for these fields?

**Review by lijunle — APPROVED** [review]

File: types/react-navigation/index.d.ts (hunk 1 above)
Comment (uniqueiniquity) [inline types/react-navigation/index.d.ts]: Waiting to merge until requested JSDoc is added.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by chagasaway** [comment]
@lijunle @uniqueiniquity just added the docs for the new fields, and sorry for the delay

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**